### PR TITLE
GKE Cluster - adding more fields

### PR DESCRIPTION
**Pull request by rambleraptor**
Getting the Ansible GKE Cluster module up-to-parity  with GKE fields

* Custom code to delete the default NodePool from a new Cluster (in line with Terraform)

* A bunch of new fields

* Logic in utils file to send blank values instead of empty dictionaries (GKE had issues with empty dictionaries)

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: products/container/api.yaml [hunk 1]
```diff
@@ -348,15 +402,77 @@ objects:
...
+                description: Whether NetworkPolicy is enabled for this cluster.
       - !ruby/object:Api::Type::String
         name: 'subnetwork'
         description: |
           The name of the Google Compute Engine subnetwork to which the cluster
           is connected.
-      # 'enableKubernetesAlpha' not supported: we are only producing GA API.
+      # TODO: figure out nodePools[]
+      - !ruby/object:Api::Type::Array
+        name: 'locations'
```
Comment (rileykarson) [inline products/container/api.yaml]: Sorry for the drive-by! The `gcloud` name for this field is _much_ more clear, `nodeLocations`. We've adopted it in Terraform as well (https://www.terraform.io/docs/providers/google/r/container_cluster.html#node_locations).

Would it be worth using this in Ansible / InSpec too?

Comment (rambleraptor) [reply products/container/api.yaml]: I can't speak to InSpec, but Ansible has a concept of aliasing, where one field has multiple names. I'll add an alias for `nodeLocations`.

Comment (SirGitsalot) [reply products/container/api.yaml]: If/when we move TF to use MM for GKE is having different names going to be a problem? And do you enjoy acronyms as much as I do?

Comment (rileykarson) [reply products/container/api.yaml]: MM overrides let us change names easily, so that's not a problem at all.
